### DISCUSSION TITLE
Query all documents matching array of sub-documents

### DISCUSSION THREAD

**Discussion by vineetdigit** [Automattic/mongoose#10945] [Q&A]
My apologies since this question is not specific to mongoose, but MongoDB.  But I have always found high-quality answers in this forum than elsewhere, hence asking it here.

I have a schema as follows
```
{
  postId: Number,
  threadId: Number,
  children: [ { postId: Number, threadId: Number } ]
}
```

I need to find all documents that match the fields of elements in the children array. 

I know there is an $in operator but it works on only single field (e.g. postId). E.g. If children array was defined as [Number] where each element is a postId, then I could use

```
model.find({postId: {$in: doc.children})
```
```

**Comment by vkarpov15** [comment]
Maybe you're looking for `$elemMatch`? I'm not sure what "all documents that match the fields of elements in the children array" means, can you please provide an example?
